tests/recordRequests.js

```diff
@@ -3,53 +3,72 @@ const fs = require(`node:fs`);
 const path = require(`node:path`);
 const v8 = require(`node:v8`);
 
-const nock = require(`nock`);
+/**
+ * @type {Map<string, {body: ArrayBuffer, status:number, headers: Record<string,string>}>}
+ */
+let mocks = new Map();
 
-const getNockFile = () =>
-  path.join(
+function getNockFile() {
+  return path.join(
     __dirname,
     `nock`,
     `${process.env.NOCK_FILE_NAME}-${process.env.RUN_CLI_ID}.dat`,
   );
-const ACCEPTED_HEADERS = new Set([`Content-Type`, `Content-Length`]);
-function filterHeaders(headers) {
-  if (!Array.isArray(headers)) return headers;
+}
 
-  const filtered = [];
-  for (let t = 0; t < headers.length; t += 2)
-    if (ACCEPTED_HEADERS.has(headers[t].toLowerCase()))
-      filtered.push(headers[t], headers[t + 1]);
+if (process.env.NOCK_ENV === `record`) {
+  const realFetch = globalThis.fetch;
+  globalThis.fetch = async (input, init) => {
+    const response = await realFetch(input, init);
+    const data = await response.arrayBuffer();
 
-  return filtered;
-}
+    const minimalHeaders = new Headers();
+    const contentType = response.headers.get(`content-type`);
+    const contentLength = response.headers.get(`content-length`);
+    if (contentType != null)
+      minimalHeaders.set(`content-type`, contentType);
+    if (contentLength != null)
+      minimalHeaders.set(`content-length`, contentLength);
 
-switch (process.env.NOCK_ENV || ``) {
-  case `record`:
-    nock.recorder.rec({
-      // eslint-disable-next-line @typescript-eslint/naming-convention
-      dont_print: true,
-      // eslint-disable-next-line @typescript-eslint/naming-convention
-      output_objects: true,
+    mocks ??= new Map();
+    mocks.set(input.toString(), {
+      body: data,
+      status: response.status,
+      headers: Object.fromEntries(minimalHeaders),
     });
 
-    process.on(`exit`, () => {
-      const nockCallObjects = nock.recorder.play();
-      for (const req of nockCallObjects)
-        if (typeof req !== `string`)
-          req.rawHeaders = filterHeaders(req.rawHeaders);
+    return new Response(data, {
+      status: response.status,
+      headers: minimalHeaders,
+    });
+  };
 
-      const serialized = v8.serialize(nockCallObjects);
+  process.once(`exit`, () => {
+    if (mocks.size) {
       fs.mkdirSync(path.dirname(getNockFile()), {recursive: true});
-      fs.writeFileSync(getNockFile(), serialized);
-    });
-    break;
+      fs.writeFileSync(getNockFile(), v8.serialize(mocks));
+    }
+  });
+} else if (process.env.NOCK_ENV === `replay`) {
+  let mocksLoaded = false;
+  globalThis.fetch = async (input, init) => {
+    if (!mocksLoaded) {
+      if (!fs.existsSync(getNockFile())) {
+        throw new Error(
+          `No nock file found for this test run; run the tests with NOCK_ENV=record to generate one`,
+        );
+      }
+
+      mocks = v8.deserialize(fs.readFileSync(getNockFile()));
```

Comment:
Using `existsSync` is both slower and less reliable (as the file could be deleted in another thread/process between the `existSync` call and the `readFileSync` call)
```suggestion
      try {
        mocks = v8.deserialize(fs.readFileSync(getNockFile()));
      } catch (err) {
        if (err?.code === 'ENOENT') {
          throw new Error(
            `No nock file found for this test run; run the tests with NOCK_ENV=record to generate one`,
            { cause: err },
          );
        }
```